libs/s25main/mapGenerator/Triangles.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2017 - 2020 Settlers Freaks (sf-team at siedler25.org)
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "mapGenerator/Triangles.h"
+
+#include <stdexcept>
+#include <vector>
+
+namespace rttr { namespace mapGenerator {
+
+    Triangle::Triangle(bool rsu, const MapPoint& position) : rsu(rsu), position(position) {}
+
+    Triangle::Triangle(bool rsu, const Position& position, const MapExtent& size)
+        : rsu(rsu), position(MakeMapPoint(position, size))
+    {}
+
+    Triangle Triangle::Inverse() const { return Triangle(!rsu, position); }
+
+    Triangle Triangle::Inverse(const MapPoint& position) const { return Triangle(!rsu, position); }
+
+    std::vector<Triangle> GetTriangles(const MapPoint& p, const MapExtent& size, Direction direction)
```

Comment:
Maybe make this a struct `TrianglePair` with named members? Or would that make the using code more complicated? Then maybe a std::array?